java/fundamentals/memory/distinction-between-stack-and-heap-memory.md

```diff
@@ -1,29 +1,59 @@
 ---
 author: adamMontgomerie
+
 type: normal
+
 category: tip
+
 tags:
   - stack
   - heap
   - memory
+
 links:
   - >-
-    [javarevisited.blogspot.jp](http://javarevisited.blogspot.jp/2013/01/difference-between-stack-and-heap-java.html){website}
----
+    [Difference Between Stack and Heap](http://javarevisited.blogspot.jp/2013/01/difference-between-stack-and-heap-java.html){website}
 
-# Distinction between stack and heap memory
+---
 
+# Distinction Between Stack and Heap Memory
 
 ---
 
 ## Content
 
-Stack and heap memory are both types of memory, but have different purposes and uses within a program:
+**Stack** and **heap** are both types of memory, but have different purposes and uses within a program.
+
+**Stack memory** is used to store local variables and method calls, while **heap memory** is used to store objects.
```

Comment:
not super clear why object variables aren't saved in the stack too 
maybe just references to the objects are put in stack?